libeos-payg/service.c

```diff
@@ -184,15 +253,35 @@ manager_new_cb (GObject      *source_object,
 {
   g_autoptr(GTask) task = G_TASK (user_data);
   EpgService *self = EPG_SERVICE (g_task_get_source_object (task));
+  g_autoptr(EpgProvider) provider = NULL;
   g_autoptr(GError) local_error = NULL;
 
-  self->provider = epg_manager_new_finish (result, &local_error);
-  if (self->provider == NULL)
+  provider = epg_manager_new_finish (result, &local_error);
+  if (provider == NULL)
     {
       g_task_return_error (task, g_steal_pointer (&local_error));
       return;
     }
 
+  epg_service_startup_complete (self, task, provider);
```

Comment:
I’d do `g_steal_pointer (&provider)` instead of just passing `provider` in order to make it explicit that this takes ownership of it. Otherwise this could look like a memory leak.